File: app/src/main/java/com/example/xpb/qingcongschool/app/MyApplication.java
package com.example.xpb.qingcongschool.app;

import android.content.res.Resources;
import android.util.DisplayMetrics;

import com.blankj.utilcode.util.Utils;
import com.example.xpb.qingcongschool.util.NetworkUtil;

import org.litepal.LitePalApplication;

/**
 * @program: QingCongSchool
 * @description: application初始化，
 * @author: 程义群
 * @create: 2018-07-08 10:21
 **/

public class MyApplication extends LitePalApplication {//继承了那个数据库的 application，以添加自己的初始化操作，然后定义到manifests
    public static final String PACKAGE_NAME = "com.example.xpb.qingcongschool";
    public static final float COURCE_LAYOUT_HEIGHT_DPI = 600f;
    private static int screenwidth = -1;
    private static float scale = 0f;
    private static int courceLayoutHeightPX = -1;

    @Override
    public void onCreate() {
        super.onCreate();
        //初始化工具类
        Utils.init(this);
        NetworkUtil.init(this);
        scale = setScale();
        screenwidth = setWidth();
        courceLayoutHeightPX = com.example.xpb.qingcongschool.util.Utils.dip2px(COURCE_LAYOUT_HEIGHT_DPI);
    }

    private int setWidth() {
        Resources resources = this.getResources();
        DisplayMetrics displayMetrics = resources.getDisplayMetrics();
        return displayMetrics.widthPixels;
    }

    private float setScale() {
        return this.getResources().getDisplayMetrics().density;
    }

    public static int getScreenwidth() {
        return screenwidth;
    }

    public static float getScale() {
        return scale;
    }

    public static int getCourceLayoutHeightPX() {
        return courceLayoutHeightPX;
    }
}

File: app/src/main/java/com/example/xpb/qingcongschool/app/MyApplication.kt
package com.example.xpb.qingcongschool.app
import com.blankj.utilcode.util.Utils


import com.facebook.drawee.backends.pipeline.Fresco
import com.facebook.stetho.Stetho
import org.litepal.LitePalApplication


/**
 * Created by ${yf} on 2017/2/16.
 */
class MyApplication : LitePalApplication() {//继承了那个数据库的 application，以添加自己的初始化操作，然后定义到manifests

    override fun onCreate() {
        super.onCreate()
        //初始化Fresco
        Fresco.initialize(this)
        //初始化工具类
        Utils.init(this)

        //初始化com.facebook.stetho:stetho
        Stetho.initializeWithDefaults(this);

        scale=getScale()
        screenwidth=getWidth()
        //statusBarHeight=getstatusBarHeight1()
        //toolbarHeight=getToolbarHeight()
        //screenHeight=getHeight()
        courceLayoutHeightPX=com.example.xpb.qingcongschool.util.Utils.dip2px(courceLayoutHeightDPI)
    }

    private fun getWidth(): Int {
        val resources = this.resources
        val displayMetrics = resources.displayMetrics
        return displayMetrics.widthPixels
    }

    private fun getScale(): Float {
        return this.resources.displayMetrics.density
    }

    /*private fun getstatusBarHeight1():Int{
        val resourceId = resources.getIdentifier("status_bar_height", "dimen", "android")
        if (resourceId > 0) {
            //根据资源ID获取响应的尺寸值
            return resources.getDimensionPixelSize(resourceId)
        }else{
            return -1
        }
    }*/

    /*private fun getToolbarHeight():Int{
        return com.example.xpb.qingcongschool.util.Utils.dip2px(56f)
    }*/

    /*private fun getHeight():Int{
        return ScreenUtils.getScreenHeight()
    }*/



    companion object {
        val packagename = "com.example.xpb.qingcongschool"
        var screenwidth = -1
        //var screenHeight = -1
        var scale = 0f
        //var statusBarHeight = -1//状态栏高度
        //var toolbarHeight = -1//toolbar高度
        val courceLayoutHeightDPI=600f//课程展示VIew的高度
        var courceLayoutHeightPX= -1
    }
}
